pkg/keyspace/tso_keyspace_group.go

```diff
@@ -101,7 +101,7 @@ func NewKeyspaceGroupManager(ctx context.Context, store endpoint.KeyspaceGroupSt
 }
 
 // Bootstrap saves default keyspace group info and init group mapping in the memory.
-func (m *GroupManager) Bootstrap() error {
+func (m *GroupManager) Bootstrap(ctx context.Context) error {
```

Comment:
I have understood the problem, and I feel this fix could be cleaner, though it can fix the problem. The GroupManage still holds the context directly inherited from pd server, but the resources are under RaftCluster's context.

With the current logic, keyspaceGroupManager's state is leader instance specific, can we new KeyspaceGroupManager inside of RaftCluster.Start with RaftCluster's context, just as NewRuleManager/NewRegionLabeler/NewReplicationModeManager does. We'll naturally avoid the bug.

Even better, can every API node, including leader and follower, keep and track keyspace group's membership/tso service registry/distribution change to serve keyspace group meta query, but only leader can serve membership/distribution write requests?

Reply:
When we new a keyspaceManger, we need to pass keyspaceGroupManager as a parameter.